store/modules/stimulation/getters.js

```diff
@@ -17,3 +16,48 @@ export default {
     if (state.new_protocol.time_unit.includes("seconds")) return "Time (s)";
   },
 };
+
+const get_new_protocol_color = ({ protocol_list }) => {
+  let check_duplicate = false;
+  const color = "#" + Math.floor(Math.random() * 16777215).toString(16);
+  protocol_list.map((protocol) => {
+    if (protocol.color === color) check_duplicate = true;
+  });
+  if (color === "#b7b7b7" || color === "#000000" || check_duplicate) get_new_protocol_color();
+  else return color;
+};
+
+const get_new_protocol_letter = ({ protocol_list }) => {
+  const current_protocol_assignment = protocol_list[protocol_list.length - 1].letter;
+  const current_alphabet_idx = alphabet.indexOf(current_protocol_assignment);
+  return alphabet[current_alphabet_idx + 1];
+};
+
+const alphabet = [
+  "A",
+  "B",
+  "C",
+  "D",
+  "E",
+  "F",
+  "G",
+  "H",
+  "I",
+  "J",
+  "K",
+  "L",
+  "M",
+  "N",
+  "O",
+  "P",
+  "Q",
+  "R",
+  "S",
+  "T",
+  "U",
+  "V",
+  "W",
+  "X",
+  "Y",
+  "Z",
+];
```

Comment:
if you wanted to shorten this up you could do something like

```javascript
const alphabet = Array.from(Array(26).keys()).map((i) => String.fromCharCode(65 + i))

Reply:
that's great, it did look funny. Changes made!